src/java/org/apache/cassandra/schema/TableMetadata.java

```diff
@@ -1283,37 +1291,16 @@ private void appendColumnDefinitions(CqlBuilder builder,
         while (iter.hasNext())
         {
             ColumnMetadata column = iter.next();
-            // If the column has been re-added after a drop, we don't include it right away. Instead, we'll add the
-            // dropped one first below, then we'll issue the DROP and then the actual ADD for this column, thus
-            // simulating the proper sequence of events.
-            if (includeDroppedColumns && droppedColumns.containsKey(column.name.bytes))
-                continue;
-
             column.appendCqlTo(builder);
 
             if (hasSingleColumnPrimaryKey && column.isPartitionKey())
                 builder.append(" PRIMARY KEY");
 
-            if (!hasSingleColumnPrimaryKey || (includeDroppedColumns && !droppedColumns.isEmpty()) || iter.hasNext())
+            if (!hasSingleColumnPrimaryKey || iter.hasNext())
                 builder.append(',');
 
             builder.newLine();
         }
-
-        if (includeDroppedColumns)
-        {
-            Iterator<DroppedColumn> iterDropped = droppedColumns.values().iterator();
-            while (iterDropped.hasNext())
-            {
-                DroppedColumn dropped = iterDropped.next();
-                dropped.column.appendCqlTo(builder);
-
-                if (!hasSingleColumnPrimaryKey || iter.hasNext())
```

Comment:
It doesn't matter, as this code is being deleted, but this is a bug - `iter.hasNext()` should be `iterDropped.hasNext()` - the wrong iterator variable is being checked and a trailing comma will never be added.

Reply:
this is how nesting iterators ends ;-)
this should be an extracted method to append dropped column => it would be not possible to make such mistake.